modules/standard/IO.chpl

```diff
@@ -5663,14 +5769,332 @@ proc _channel.readLine(ref b: bytes,
 
   :throws UnexpectedEofError: Thrown if unexpected EOF encountered while reading.
   :throws SystemError: Thrown if data could not be read from the channel.
-  :throws IoError: Thrown if the line is longer than `maxSize`. It leaves the input marker at the beginning of the offending line.
+  :throws IoError: Thrown if the line is longer than `maxSize`. It leaves the fileReader position at the beginning of the offending line.
 */
 proc _channel.readLine(type t=string, maxSize=-1, stripNewline=false): t throws where t==string || t==bytes {
   var retval: t;
   this.readLine(retval, maxSize, stripNewline);
   return retval;
 }
 
+
+/*
+  Read until the given separator is found, returning the contents of the
+  ``fileReader`` through that point.
+
+  If the separator is found, the ``fileReader`` position is left immediately
+  after it. If the separator could not be found in the next ``maxSize``
+  bytes, a ``BadFormatError`` is thrown and the ``fileReader``'s
+  position is not changed. If EOF is reached before finding the separator,
+  the remainder of the ``fileReader``'s contents are returned and the position
+  is left at EOF.
+
+  To match with multiple separators, or a more complex separator, use the
+  overload of :proc:`~Regex.fileReader.readThrough` that accepts a
+  :type:`~Regex.regex` separator.
+
+  :arg separator: The separator to match with. Must be a :type:`~String.string`
+    or :type:`~Bytes.bytes`.
+  :arg maxSize: The maximum number of bytes to read. For the default value of
+    ``-1``, this method will read until EOF.
+  :arg stripSeparator: Whether to strip the separator from the returned
+    ``string`` or ``bytes``. If ``true``, the returned value will not include
+    the separator.
+  :returns: A ``string`` or ``bytes`` with the contents of the ``fileReader``
+    up to (and possibly including) the separator.
+
+  :throws EofError: Thrown if nothing could be returned (i.e., the
```

Comment:
How about

```
:throws EofError: if nothing could be read because the ``fileReader`` was already at EOF.
```


If the separator was the only data left in the file, I would expect it to return that separator (if stripSeparator=false) or the empty string (if stripSeparator=true).